Add otelcol.receiver.solace (#2084)
grafana/alloy: docs/sources/reference/components/otelcol/otelcol.receiver.solace.md

---
canonical: https://grafana.com/docs/alloy/latest/reference/components/otelcol/otelcol.receiver.solace/
description: Learn about otelcol.receiver.solace
title: otelcol.receiver.solace
---

> # otelcol.receiver.solace

Reviewer: Is it becoming stable right away?
I think you want to make it exprimental / public preview first?

Author: discussed internally

`otelcol.receiver.solace` accepts traces from a [Solace PubSub+ Event Broker](https://solace.com/products/event-broker/) and
forwards it to other `otelcol.*` components.

> > **NOTE**: `otelcol.receiver.solace` is a wrapper over the upstream
> > OpenTelemetry Collector `solace` receiver from the `otelcol-contrib`
> > distribution. Bug reports or feature requests will be redirected to the
> > upstream repository, if necessary.

Reviewer 2: ```suggestion
{{< admonition type="note" >}}
`otelcol.receiver.solace` is a wrapper over the upstream OpenTelemetry Collector `solace` receiver from the `otelcol-contrib` distribution.
Bug reports or feature requests will be redirected to the upstream repository, if necessary.
{{< /admonition >}}
```

> Multiple `otelcol.receiver.solace` components can be specified by giving them
> different labels.

Reviewer 2: ```suggestion
You can specify multiple `otelcol.receiver.solace` components by giving them different labels.
```

## Usage

```alloy
otelcol.receiver.solace "LABEL" {
  queue = "QUEUE"
  auth {
    // sasl_plain or sasl_xauth2 or sasl_external block
  }
  output {
    traces  = [...]
  }
}
```

## Arguments

The following arguments are supported:

| Name                 | Type     | Description                                                               | Default          | Required |
| -------------------- | -------- | ------------------------------------------------------------------------- | ---------------- | -------- |
| `queue`              | `string` | Name of the Solace telemetry queue to get span trace messages from.       |                  | yes      |

> | `broker`             | `string` | Name of the Solace broker using amqp over tls.                            | `localhost:5671` | no       |

Reviewer 3: Should we consider keeping this the same as upstream, so when upstream is changed to utilize multiple broker support we don't need to change the config in Alloy?

Author: To support multiple brokers, they would need to change the structure of the config because with several brokers you probably want the possibility to specify a queue and credentials for each broker. Also they should change the name to "brokers" for consistency and change the doc since they refer to it as "the broker" + the component is already "old" and considered "stable" in otel standard. So I don't see this change happening in the near future / happening without other breaking changes in the config.
I would prefer to ask upstream if they are ok to switch to a string and if not, whether they are really planning to support multiple brokers at some point

Author: Upstream contribution: https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/36387

> | `broker`             | `string` | Name of the Solace broker using amqp over tls.                            | `localhost:5671` | no       |

Reviewer: ```suggestion
| `broker`             | `string` | Name of the Solace broker using AMQP over TLS.                            | `localhost:5671` | no       |
```

| `max_unacknowledged` | `int`    | Maximum number of unacknowledged messages the Solace broker can transmit. | 10               | no       |

> `queue` should have the format `queue://#telemetry-myTelemetryProfile`.

Reviewer 2: Should or must? Should implies it can have some other format too.

Author: "must" is better then

Author: I updated it

## Blocks

The following blocks are supported inside the definition of
`otelcol.receiver.solace`:

| Hierarchy                      | Block              | Description                                                                                                                      | Required |
| ------------------------------ | ------------------ | -------------------------------------------------------------------------------------------------------------------------------- | -------- |
| authentication                 | [authentication][] | Configures authentication for connecting to the Solace broker.                                                                   | yes      |

> | authentication > sasl_plain    | [sasl_plain][]     | Authenticates against Solace broker with SASL PLAIN.                                                                             | no       |
> | authentication > sasl_xauth2   | [sasl_xauth2][]    | Authenticates against Solace broker with SASL XOauth2.                                                                           | no       |
> | authentication > sasl_external | [sasl_external][]  | Authenticates against Solace broker with SASL External.                                                                          | no       |

Reviewer 2: ```suggestion
| authentication > sasl_plain    | [sasl_plain][]     | Authenticates against the Solace broker with SASL PLAIN.                                                                         | no       |
| authentication > sasl_xauth2   | [sasl_xauth2][]    | Authenticates against the Solace broker with SASL XOauth2.                                                                       | no       |
| authentication > sasl_external | [sasl_external][]  | Authenticates against the Solace broker with SASL External.                                                                      | no       |
```
